File: 0387-first-unique-character-in-a-string-字符串中的第一个唯一字符/0387.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        table = {}
        i = 0
        for c in s:
            if c not in table:
                table[c] = i
            else: 
                table[c] = -1
            i += 1

        for key, value in table.items():
            if value >= 0:
                return value

        return -1

File: 0387-first-unique-character-in-a-string-字符串中的第一个唯一字符/0387.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        vector<int> table(26, -1);
        vector<bool> uniqS(s.size(), false);

        for (int i = 0; i != s.size(); ++i) {
            if (table[s[i] - 'a'] == -1) {
                table[s[i] - 'a'] = i;
            } else {
                uniqS[i] = true;
                uniqS[table[s[i] - 'a']] = true;
            }
        }

        for (int i = 0; i != uniqS.size(); ++i) {
            if (uniqS[i] == false) {
                return i; 
            }
        }

        return -1;
    }
};